client/request/select.go

```diff
@@ -107,3 +109,88 @@ func (s *Select) validateGroupBy() []error {
 
 	return result
 }
+
+// selectJson is a private object used for handling json deserialization
+// of `Select` objects.
+type selectJson struct {
+	Field
+	DocKeys     immutable.Option[[]string]
+	CID         immutable.Option[string]
+	Root        SelectionType
+	Limit       immutable.Option[uint64]
+	Offset      immutable.Option[uint64]
+	OrderBy     immutable.Option[OrderBy]
+	GroupBy     immutable.Option[GroupBy]
+	Filter      immutable.Option[Filter]
+	ShowDeleted bool
+
+	// Properties above this line match the `Select` object and
+	// are deserialized using the normal/default logic.
+	// Properties below this line require custom logic in `UnmarshalJSON`
+	// in order to be deserialized correctly.
```

Comment:
suggestion: Although I like this comment, I find that overtime this might be hard to enforce / maintain. Perhaps in addition to this add a different prefix to variable names to the ones that require custom logic? i.e. `CustomFields` instead of `Fields`

Reply:
The current solution for sure has it's flaws, but I'm not sure I like that suggestion.  I'll add a task to re-visit and see how I feel about it again in a little bit :)

- [ ] Revisit this suggestion

Reply:
I still don't like the suggestion, and will leave as-is unless anyone continues this thread :)